scripts/main.js

```diff
@@ -0,0 +1,250 @@
+    gameBoard.style.display = 'grid';
+    gameBoard.className = 'fade'
+    gameBoard.style.zIndex = '1';
+    status.style.display = 'flex';
+    status.className = 'fade';
+    start.disabled = true;
+    player = 'x';
+    textUpdate.textContent = `It's ${playerXName}'s turn!`;
+    cells.forEach((cell) => {
+        cell.addEventListener('click', clicked);
+        cell.style.backgroundColor = '#f0f0f0';
+    })
+}
+//on click function for cell divs.  All basic game logic is included here.
+function clicked() {
+    turnCount += 1;
+    event.target.textContent = player.toUpperCase();
+    event.target.removeEventListener('click', clicked);
+    board[event.target.id] = player;
+    state = buildBoardState(board);
+    let win = checkWin(state, player);
+    if (win === true) {
+        textUpdate.textContent = `${player === 'x' ? playerXName : playerOName} WINS!!!`
+        return reset();
+    } else {
+        if (turnCount === 9) {
+            textUpdate.textContent = `It's a draw!`
+            return reset();
+        }
+        player = player === 'x' ? 'o' : 'x';
+        if (twoPlayer === true) {
+            return textUpdate.textContent = `It's ${player === 'x' ? playerXName : playerOName}'s turn!`
+        } else if (onePlayer === true) {
+            textUpdate.textContent = `It's ${player === 'x' ? playerXName : playerOName}'s turn!`
+            state = buildBoardState(board);
+            let bestMove = makeMove(state, player)
+            cells[bestMove].textContent = player.toUpperCase();
+            cells[bestMove].removeEventListener('click', clicked);
+            board[bestMove.toString()] = player;
+            state = buildBoardState(board);
+            let aiWin = checkWin(state, player);
+            if (aiWin === true) {
+                textUpdate.textContent = `${player === 'x' ? playerXName : playerOName} WINS!!!`
+                return reset();
+            } else {
+                turnCount += 1;
+                player = player === 'x' ? 'o' : 'x';
+                return textUpdate.textContent = `It's ${player === 'x' ? playerXName : playerOName}'s turn!`
+            }
+        }
+    }
+}
+//determines if there is a win on the board
+function checkWin(boardState, currentPlayer) {
+    let player = currentPlayer === 'x' ? 'playerX' : 'playerO';
+    for (let win of winningArrays) {
+        let winCount = 0;
+        for (let coordinate of win) {
+            if (boardState[player].includes(coordinate)) {
+                winCount += 1;
+            }
+        }
+        if (winCount >= 3) {
+            for (let coordinate of win) {
+                cells[coordinate].style.backgroundColor = 'darkgray'
+            }
+            return true;
+        }
+    }
+    return false;
+}
+//pads numbers with a zero for the clock if they are single digit
+function pad(number) {
+    if ((number + '').length === 2) {
+        return number + '';
+    } else {
+        return '0' + number;
+    }
+}
```

Comment:
I was wondering where the `pad` function was coming from, and now I see it was here all along. Instead of adding the number to a string you could've called the `toString()` method on your `number` variable.

Reply:
Is this better for readability? I was mostly just trying to save a few characters.